ratis-server/src/main/java/org/apache/ratis/server/impl/ServerImplUtils.java

```diff
@@ -47,14 +50,12 @@ public static RaftServerProxy newRaftServer(
       RaftPeerId id, RaftGroup group, StateMachine.Registry stateMachineRegistry,
       RaftProperties properties, Parameters parameters) throws IOException {
     RaftServer.LOG.debug("newRaftServer: {}, {}", id, group);
-    final RaftServerProxy proxy = newRaftServer(id, stateMachineRegistry, properties, parameters);
-    proxy.initGroups(group);
-    return proxy;
-  }
 
-  private static RaftServerProxy newRaftServer(
-      RaftPeerId id, StateMachine.Registry stateMachineRegistry, RaftProperties properties, Parameters parameters)
-      throws IOException {
+    if (group != null && !group.getPeers().isEmpty()) {
+      final Set<RaftPeerId> ids = group.getPeers().stream().map(RaftPeer::getId).collect(Collectors.toSet());
```

Comment:
Use RaftGroup.getPeer(id) instead of creating a set.

Reply:
Thanks! I don't notice this method :P